zingo-testutils/src/assertions.rs

```diff
@@ -60,3 +60,20 @@ pub async fn assert_send_outputs_match_recipient<NoteId>(
         );
     }
 }
+
+/// assert that proposing shield causes a specific error
+pub async fn assert_cant_shield(client: &zingolib::lightclient::LightClient) {
+    assert!(match client.propose_shield().await.unwrap_err() {
+            zingolib::lightclient::propose::ProposeShieldError::Component(zcash_client_backend::data_api::error::Error::NoteSelection(zcash_client_backend::data_api::wallet::input_selection::GreedyInputSelectorError::Change(zcash_client_backend::fees::ChangeError::InsufficientFunds { available, required }))) => {
+                println!("available {}, required more than {}", available.into_u64(), required.into_u64());
+                true
+            },
+            zingolib::lightclient::propose::ProposeShieldError::BelowThreshold(amount) => {
+                println!("possible note {}, threshold {}", amount, zingolib::lightclient::propose::SHIELDING_CUTOFF);
+                true},
```

Comment:
I think asserting that a specific expected error is the correct way to test error cases, this test will pass when the shield fails in a way that wasnt expected.

i like the pattern:
assert!(matches!(client.propose_shield().await, Err(zingolib::lightclient::propose::ProposeShieldError::BelowThreshold(_))))

this way we dont need to waste time trying to write the error messages out and fixing all tests when the message changes etc.

this is my preference but I think tests are more readable when this assert is directly in the test itself rather than helperised as this format is quite concise anyway